stock_picking_group_by_partner_by_carrier/models/stock_move.py

```diff
@@ -114,8 +111,11 @@ def _domain_search_picking_handle_move_type(self):
         By default the move type is taken from the procurement group.
         Override to customize this behavior.
         """
-        # avoid mixing picking policies
-        return [("move_type", "=", self.group_id.move_type)]
+        if self.group_id.move_type:
+            # avoid mixing picking policies
+            return [("move_type", "=", self.group_id.move_type)]
```

Comment:
if the move type is missing on the procurement group, it now defaults to the one configured on the picking type. Otherwise, we can assume direct as default.
```suggestion
        group = self.group_id.move_type or self.picking_type_id.move_type or "direct"
        return [("move_type", "=", self.group_id.move_type)]
```